### ISSUE TITLE
perf(solve): sorted-index hints and precomputed inverse diagonals in the frozen multigrid apply

### ISSUE THREAD

**Issue by ctdegroot** [DeGrootResearchGroup/aquaflux#110]
Surfaced by a code-wide speed review. Both items are in the innermost preconditioner kernel.

## 1. `_coo_apply` never tells XLA the segments are sorted

`aquaflux/solve/multigrid.py:331-336`:

```python
return segment_sum(val * x[col], row, n_out)
```

No `indices_are_sorted` / `unique_indices`. Without the hint this lowers to a generic `scatter-add` that XLA must assume is arbitrarily permuted; with `indices_are_sorted=True` it can emit a segmented reduce (on GPU, a sorted segmented reduction instead of atomics).

This is the innermost kernel of the entire preconditioner. Measured on a 1024-cell Poisson hierarchy (levels 1024 -> 176 -> 22 -> 4), one V-cycle compiles to **24 scatter ops**, and one V-cycle runs per GMRES iteration per Newton step.

The segment ids were checked per level:

```
level 0  A.row sorted: True   P.frow (prolong) sorted: True   P.ccol (restrict) sorted: False
level 1  A.row sorted: True   P.frow (prolong) sorted: True   P.ccol (restrict) sorted: False
level 2  A.row sorted: True
```

So `_operator_matvec` (22 of the 24) and `prolong` can take the hint as-is. `restrict` transposes the COO in place (`_smoothed_ops`, line 442) and loses sortedness — pre-sorting that triple once at build time in `_sparse_level` / `_air_level` makes it qualify too.

Matters most on GPU, and scales with mesh size.

## 2. Frozen level diagonals are re-inverted on every smoother apply

`aquaflux/solve/multigrid.py:579-580` (`_jacobi_smooth`) and `:905` (`_fc_jacobi`) compute `1.0 / level.diagonal` per call, and `_fc_jacobi` additionally recomputes `omega * level.f_mask * inv_diagonal` per sweep. The diagonal and the masks are build-time frozen constants.

XLA will CSE within a trace so the runtime cost is small, but storing `inv_diagonal` — and the pre-fused `omega * mask * inv_diagonal` — on `_SparseLevel` / `_AirLevel` at build time is free and removes n-element divides from the hot graph.

## Note — checked, not a problem

`_frozen_v_cycle` pre-smooths from `jnp.zeros_like(b)` and `_fixed_cycle_solve` starts from zero, so four `A @ zeros` matvecs look wasted. XLA constant-folds all of them: the jaxpr has 28 scatter-adds, the optimized HLO has 24 — exactly the 4 zero-guess matvecs (3 levels + outer). No change needed there.

## Acceptance

- Optimized HLO for one V-cycle shows segmented reduces rather than generic scatter-adds where the indices are sorted.
- No n-element divide in the smoother bodies.
- V-cycle contraction and the existing multigrid tests unchanged.
